vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -78,6 +78,14 @@ def verify_api_credentials(options={})
     rescue Excon::Errors::Unauthorized => err
       $log.error("MIQ(#{self.class.name}.verify_api_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqEVMLoginError, "Login failed due to a bad username or password."
+    rescue Excon::Errors::Timeout => err
+      $log.error("MIQ(#{self.class.name}.verify_api_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
+      raise MiqException::MiqEVMLoginError, "Login attempt timed out"
```

Comment:
I looked into what is used elsewhere, I don't see much consistency but tried to use more proper exceptions.

Either way, letting it bubble up would definitely be easier, we would get rid of any "Unexpected error" messages, but possibly to the detriment of user-friendlyness, so I don't know..